lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -624,6 +624,7 @@ defmodule ExUnit.Case do
 
     moduletag = Module.get_attribute(mod, :moduletag)
     tag = Module.delete_attribute(mod, :tag)
+    {async, _parameterize} = Module.get_attribute(mod, :ex_unit_module)
```

Comment:
Instead of reading it here, maybe we can make it part of `moduletag` and that will just work? Then I think we can also remove it from `__before_compile__` too.

Reply:
Another option is to set it inside ExUnit.Runner in run_module/prepare_tests.

Reply:
We validate that the module tags don't set any reserved tags. Async is in the list of reserved tags. If we remove it there and do a `Module.put_attribute(module, :moduletag, [async: async?])` in `__register__` it would work, but then users can also do

```elixir
defmodule MyTest do
  use ExUnit.Case

  @moduletag async: true
end
```

I don't know if that's fine. The current change is basically how it was previously and I'm really no expert on ExUnit internals :)